app/helpers/application_helper.rb

```diff
@@ -248,7 +248,9 @@ def required_helper
 
 
   def user_for_comment(comment)
-    if comment.user.nil?
+    if comment.user == current_user
```

Comment:
This must be the 2nd check. Otherwise a guest user (meaning `current_user` == `nil`) could be shown as "You" when the comment doesn't have a user anymore.